frontend/packages/console-dynamic-plugin-sdk/src/api/k8s-types.ts

```diff
@@ -0,0 +1,151 @@
+import {
+  K8sResourceCommon,
+  CustomResourceDefinitionKind,
+  Selector,
+  OwnerReference,
+  Options,
+  Patch,
+} from '../extensions/console-types';
+import { K8sKind, ExtensionK8sGroupModel } from './common-types';
+
+// TODO - Convert WS factory to typescript and remove this temporary type
+type WSFactory = (id: any, options: any) => void;
+
+/**
+ * K8s CRUD utility types to be exposed by dynamic plugin SDK.
+ *  */
+
+export type K8sGet = (
+  kind: K8sKind,
+  name: string,
+  ns?: string,
+  opts?: Options,
+  requestInit?: RequestInit,
+) => Promise<any>;
+
+export type K8sCreate = <R extends K8sResourceCommon>(
+  kind: K8sKind,
+  data: R,
+  opts?: Options,
+) => Promise<any>;
+
+export type K8sUpdate = <R extends K8sResourceCommon>(
+  kind: K8sKind,
+  data: R,
+  ns?: string,
+  name?: string,
+  opts?: Options,
+) => Promise<R>;
+
+export type K8sPatch = <R extends K8sResourceCommon>(
+  kind: K8sKind,
+  resource: R,
+  data: Patch[],
+  opts?: Options,
+) => Promise<any>;
+
+export type K8sPatchByName = (
+  kind: K8sKind,
+  name: string,
+  namespace: string,
+  data: Patch[],
+  opts?: Options,
+) => Promise<any>;
+
+export type K8sKill = <R extends K8sResourceCommon>(
+  kind: K8sKind,
+  resource: R,
+  opts?: Options,
+  requestInit?: RequestInit,
+  json?: Record<string, any>,
+) => Promise<any>;
+
+export type K8sKillByName = <R extends K8sResourceCommon>(
+  kind: K8sKind,
+  name: string,
+  namespace?: string,
+  opts?: Options,
+  requestInit?: RequestInit,
+) => Promise<R>;
+
+export type K8sList = (
+  kind: K8sKind,
+  params?: {
+    [key: string]: any;
+  },
+  raw?: boolean,
+  requestInit?: RequestInit,
+) => Promise<any>;
+
+export type K8sListPartialMetadata = (
+  kind: K8sKind,
+  params?: {
+    [key: string]: any;
+  },
+  raw?: boolean,
+) => Promise<any>;
+
+export type K8sWatch = (
+  kind: K8sKind,
+  query?: {
+    labelSelector?: Selector;
+    resourceVersion?: string;
+    ns?: string;
+    fieldSelector?: string;
+  },
+  wsOptions?: {
+    [key: string]: any;
+  },
+) => WSFactory;
+
+export type K8sWaitForUpdate = <R extends K8sResourceCommon>(
+  kind: K8sKind,
+  resource: R,
+  checkCondition: (kind: R) => boolean,
+  timeoutInMs: number,
+) => Promise<unknown>;
+
+/**
+ * K8s reference utility types to be exposed by dynamic plugin SDK.
```

Comment:
Nit: I think it's better to represent this as non-block comment via `// ..` syntax.